extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/InterfaceConfigPropertiesUtil.java

```diff
@@ -134,12 +135,32 @@ static String generateImplementationForInterfaceConfigProperties(ClassInfo origi
 
                             Type genericType = determineSingleGenericType(returnType,
                                     method.declaringClass().name());
-                            ResultHandle result = methodCreator.invokeInterfaceMethod(
-                                    MethodDescriptor.ofMethod(Config.class, "getOptionalValue", Optional.class, String.class,
-                                            Class.class),
-                                    config, methodCreator.load(fullConfigName),
-                                    methodCreator.loadClass(genericType.name().toString()));
-                            methodCreator.returnValue(result);
+
```

Comment:
@geoand no way to factorize this with the `ClassConfigPropertiesUtil`?

Reply:
Sure, do you want to do that as an exercise to get your hands dirty with this code?

Reply:
It looks like a trap 🤔

Reply:
Sure